chain/ethevents/handlers.go

```diff
@@ -364,3 +379,33 @@ func namespaceOracleRemovedMeta(
 	log.Debugf("namespaceOracleRemoved eventData: %+v", structuredData)
 	return ph.RemoveOracleTxArgs(ctx, structuredData.OracleAddress, structuredData.Namespace)
 }
+
+func checkEthereumTxCreator(ctx context.Context,
+	txHash common.Hash,
+	blockHash common.Hash,
+	txIndex uint,
+	ethereumWhiteList map[common.Address]bool,
+	ethclient *ethclient.Client,
+) error {
+	// get tx from ethereum
+	tx, _, err := ethclient.TransactionByHash(ctx, txHash)
+	if err != nil {
+		return fmt.Errorf("cannot fetch tx by hash: %w", err)
+	}
+	if tx != nil {
+		// get from
+		sender, err := ethclient.TransactionSender(ctx, tx, blockHash, txIndex)
+		if err != nil {
+			return fmt.Errorf("cannot fetch tx sender: %w", err)
+		}
+		// check from is whitelisted
+		if val, ok := ethereumWhiteList[sender]; !ok {
+			return fmt.Errorf("recovered address not in ethereum whitelist")
+		} else {
+			if val {
+				return nil
+			}
+		}
+	}
+	return fmt.Errorf("obtained tx from Ethereum is nil")
```

Comment:
as said before, I don't think this can ever happen since you already checked the error.